engines/made/made.cpp

```diff
@@ -156,6 +204,137 @@ void MadeEngine::resetAllTimers() {
+		ttsMan->stop();
+	}
+}
+
+#ifdef USE_TTS
+
+void MadeEngine::checkHoveringSaveLoadScreen() {
+	if ((_saveScreenOpen || _loadScreenOpen) && getGameID() == GID_RTZ) {
+		bool hoveringOverButton = false;
+		for (uint8 i = 0; i < kRtzSaveLoadButtonCount; ++i) {
+			if (rtzSaveLoadScreenButtons[i].contains(_eventMouseX, _eventMouseY)) {
+				if (_previousRect != i) {
+					if (i == kTextBox) {
+						int index = MIN((_eventMouseY - 20) / kRtzSaveBoxHeight, 9);
+
+						if (index != _previousTextBox) {
+							// The string of object 1946 is the number of the last save slot
+							const char *lastSaveSlot = _dat->getObjectString(1946);
+							if (lastSaveSlot) {
+								// Sometimes, string 1946 is not yet updated to the number of the last save slot at this time, so
+								// round to the next multiple of 10 to keep the voicing accurate
+								sayText(Common::String::format("%d", ((atoi(lastSaveSlot) + 9) / 10) * 10 - 9 + index));
+							}
+							_previousTextBox = index;
+						}
+					} else {
+						const char *text = nullptr;
+
+						MenuResource *menu = _res->getMenu(1);
+						if (menu) {
+							if (i == kCancel) {
+								text = menu->getString(0);
+							} else {
+								if (_saveScreenOpen) {
+									text = menu->getString(26);
+								} else {
+									text = menu->getString(27);
+								}
+							}	
+						}
+
+						if (text) {
+							sayText(text);
+						}
+
+						_previousRect = i;
+					}
+				}
+
+				hoveringOverButton = true;
+				break;
+			}
+		}
+
+		if (!hoveringOverButton) {
+			_previousRect = -1;
+			_previousTextBox = -1;
+		}
+	}
+}
+
+void MadeEngine::checkHoveringPlayOMatic() {
+	if (_saveScreenOpen && getGameID() == GID_LGOP2) {
+		bool hoveringOverButton = false;
+		for (uint8 i = 0; i < kLgop2PlayOMaticButtonCount; ++i) {
+			if (lgop2PlayOMaticButtons[i].contains(_eventMouseX, _eventMouseY)) {
+				if (_previousRect != i) {
+					const char *text = _dat->getString(_playOMaticButtonTextIndices[i]);
+					if (text) {
+						sayText(text, Common::TextToSpeechManager::INTERRUPT);
+					}
+
+					_previousRect = i;
+				}
+
+				hoveringOverButton = true;
+				break;
+			}
+		}
+
+		if (!hoveringOverButton) {
+			_previousRect = -1;
+		}
+	}
+}
+
+void MadeEngine::voiceHighlightedPlayOMaticButton(int spriteY) const {
+	for (uint8 i = 0; i < kLgop2PlayOMaticButtonCount; ++i) {
+		if (lgop2PlayOMaticButtons[i].top == spriteY) {
+			const char *text = _dat->getString(_playOMaticButtonTextIndices[i]);
+			if (text) {
+				sayText(text, Common::TextToSpeechManager::INTERRUPT);
+			}
+
+			return;
+		}
+	}
+}
+
+bool MadeEngine::tryAddPlayOMaticButtonTextIndex(int16 textIndex) {
+	if (_playOMaticButtonIndex >= kLgop2PlayOMaticButtonCount) {
+		return false;
+	}
+
+	for (uint8 i = 0; i < kLgop2PlayOMaticTextCount; ++i) {
+		if (lgop2PlayOMaticTextIndices[i] == textIndex) {
+			_playOMaticButtonTextIndices[_playOMaticButtonIndex] = textIndex;
+			_playOMaticButtonIndex++;
+			return true;
+		}
+	}
+
```

Comment:
This function does not work for me because the text indices in my version are different from the one in `lgop2PlayOMaticTextIndices`

I have (French version) 14851, 15596, 15599, 15605, 15611, 15617, 15624, 15631, 15635, 15641, 15646, 15652, 15656, 15660, 15667,  15673, 15681, 15687, 15690, 15695, 15701.

I suspect these are not the only text indices that differ. Using those to identify the text in various places seems a bit fragile. But I don't really have a better idea. Maybe it would be better to check we are on the PlayOMatic screen and what the `_textX` and `_textY` values are instead of relying on the text indices?

Reply:
It might be better. Currently none of the code that relies on the text indices is working for me in LGOP.

For example:
> // 13484 is the "over a decade of the best interactive fiction" line in the beginning, which has no voiceover
> // 14185 is the "choose your character" message, and 13876 is the copyright message in the intro

Those are 14851 and 15589 respectively for me. So that don't get voiced currently.